### PR TITLE
[WIP] Nested virt bool

### LINKED ISSUE #17: Add a nested_virt option
Add a nested_virt param to the kubevirt-apb.  This is going to require running:
```
echo "options kvm-intel nested=1" > /etc/modprobe.d/kvm-intel.conf
rmmod kvm-intel && modprobe kvm-intel || true
cat /sys/module/kvm_intel/parameters/nested
```

In order for this to work, the container needs to be a root user and needs to mount in ```/lib/modules```.

### PR DESCRIPTION
**Pull request by rthallisey**
depends-on: https://github.com/openshift/ansible-service-broker/issues/812

fixes: #17

### CODE REVIEW

File: roles/kubevirt-apb/tasks/provision.yml [hunk 1]
```diff
@@ -10,6 +10,24 @@
     - kubevirt-privileged
     - kubevirt-controller
 
+- name: Enable Nested Virt
+  shell: echo "options kvm-intel nested=1" > /etc/modprobe.d/kvm-intel.conf
+  when: nested_virt==True
+
+- name: Enable Nested Virt
+  shell: rmmod kvm-intel && modprobe kvm-intel || true
```
Comment (mareklibra) [inline roles/kubevirt-apb/tasks/provision.yml]: What about AMD?

File: roles/kubevirt-apb/tasks/provision.yml [hunk 2]
```diff
@@ -10,6 +10,24 @@
...
+  when: nested_virt==True
+
+- name: Check Nested Virt is Enabled
+  shell: cat /sys/module/kvm_intel/parameters/nested
+  register: result
```
Comment (mareklibra) [inline roles/kubevirt-apb/tasks/provision.yml]: If successful, single letter `Y` is expected to be printed out here

**Review by mareklibra — CHANGES_REQUESTED** [review]
Nice :-)

I think we should support AMD as well.
Verification step should be improved.

**Comment by rthallisey** [comment]
@mareklibra This may be out of scope of an apb because it involves making changes to a host machine.  I created https://github.com/openshift/ansible-service-broker/issues/812 so there's a discussion around this started, but in the near term this probably makes more sense to do outside of an apb.  If https://github.com/openshift/ansible-service-broker/issues/812 is merged later on, we can always come back and add this.

**Comment by lukas-bednar** [comment]
I was dealing with nested vt on our infra too, and [here](https://galaxy.ansible.com/lukas-bednar/ansible-role-nested-virtualization) is how I deal with that, you could use there as well if you find it handy:

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Pull request closed without merging**